dojo/tools/popeye/parser.py

```diff
@@ -0,0 +1,69 @@
+import hashlib
+import json
+from dojo.models import Finding
+
+
+class PopeyeParser(object):
+    """
+    Popeye is a kubernetes cluster resource analyzer.
+    """
+
+    def get_scan_types(self):
+        return ["Popeye Scan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "Popeye Scan"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Popeye report file can be imported in JSON format (option --json)."
+
+    def get_findings(self, file, test):
+        data = json.load(file)
+
+        dupes = dict()
+        for sanitizer in data['popeye']['sanitizers']:
+            try:
+                for issue_group, issue_list in sanitizer['issues'].items():
+                    for issue in issue_list:
+                        if issue['level'] != 0:
+                            title = sanitizer['sanitizer'] + " " + issue_group + " " + issue['message']
+                            severity = self.get_defect_dojo_severity(issue['level'])
+                            description = "**Sanitizer** : " + sanitizer['sanitizer'] + "\n\n" + \
+                                        "**Resource** : " + issue_group + "\n\n" + \
+                                        "**Group** : " + issue['group'] + "\n\n" + \
+                                        "**Severity** : " + self.get_popeye_level_string(issue['level']) + "\n\n" + \
+                                        "**Message** : " + issue['message']
+
+                            finding = Finding(
+                                title=title,
+                                test=test,
+                                description=description,
+                                severity=severity,
+                                static_finding=False,
+                                dynamic_finding=True
+                            )
+                            # internal de-duplication
+                            dupe_key = hashlib.sha256(str(description + title).encode('utf-8')).hexdigest()
+                            if dupe_key not in dupes:
+                                dupes[dupe_key] = finding
+                        else:
+                            continue
+            except KeyError:
+                continue
```

Comment:
I did it that way since there were some sanitizers that does not include issues in the popeye report and was throwing the exception. I'll add a commit to avoid going on the catching exceptions way.